internal/backend/backend.go

```diff
@@ -131,3 +138,90 @@ type resource struct {
 	ScannedAt        metav1.Time   `json:"scanned_at"`
 	DeletedAt        *metav1.Time  `json:"deleted_at,omitempty"`
 }
+
+type upsertFailures struct {
+	failures map[types.UID]*upsertFailure
```

Comment:
Will this be accessed concurrently? I think it will have to be, since we'll probably need to serve metrics from a new goroutine.

We might need to change this to a sync.Map or use a mutex.

Reply:
I think the backend can handle requests concurrently, so yeah this will also need need a mutex or a sync.Map...good catch!  (we are using `-race` when testing, so that should've showed up as well, _probably_).